src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs

```diff
@@ -163,7 +176,68 @@ internal void PerformBackup()
         internal BackupConfigInfo GetBackupConfigInfo(string databaseName)
         {
             return this.backupUtilities.GetBackupConfigInfo(databaseName);
-        }   
-        
+        }
+
+        /// <summary>
+        /// Create a backup task for execution and cancellation
+        /// </summary>
+        /// <param name="sqlTask"></param>
+        /// <returns></returns>
+        internal async Task<TaskResult> BackupTask(SqlTask sqlTask)
+        {                           
+            return await await Task.Factory.StartNew(async () =>
+            {
+                sqlTask.AddMessage("In progress", SqlTaskStatus.InProgress, true);
+
+                // create a task to perform backup
+                Task<TaskResult> backupTask = Task.Factory.StartNew(() =>
+                {
+                    TaskResult result = new TaskResult();
+                    try
+                    {
+                        this.backupUtilities.PerformBackup();
+                        result.TaskStatus = SqlTaskStatus.Succeeded;
+                    }
+                    catch (Exception ex)
+                    {
+                        result.TaskStatus = SqlTaskStatus.Failed;
+                        result.ErrorMessage = ex.Message;
+                    }
+                    return result;
+                });
+
+                // create a task for backup cancellation request
+                Task<TaskResult> cancelTask = Task.Factory.StartNew(() =>
+                {
+                    TaskResult result = new TaskResult();
+                    while (true)
+                    {
+                        if (sqlTask.IsCancelRequested)
+                        {
+                            break;
+                        }
+                    };
+
+                    try
+                    {
+                        this.backupUtilities.CancelBackup();
+                        result.TaskStatus = SqlTaskStatus.Canceled;                        
+                    }
+                    catch (Exception ex)
+                    {
+                        result.TaskStatus = SqlTaskStatus.Failed;
+                        result.ErrorMessage = ex.Message;
+                    }
+
+                    return result;
+                });
+
+                Task<TaskResult> completedTask = await Task.WhenAny(backupTask, cancelTask);
```

Comment:
The signature of WhenAny is:
public static Task<Task<TResult>> WhenAny<TResult>(params Task<TResult>[] tasks);

That's why when it awaits, it gets Task<TResult>